### ISSUE TITLE
bootstrap of a new node (replace_with_repair) hangs with no clear reason

### ISSUE THREAD

**Issue by fgelcer** [scylladb/scylla#6248]
*Installation details*
Scylla version (or git commit hash): 4.0.rc1-0.20200418.21ad12669a9 with build-id f881cad8699bc4699f7c4f4c408edef91d337044
Cluster size: 4
OS (RHEL/CentOS/Ubuntu/AWS AMI): ami-09f702affc5cb6499

at `2020-04-20 03:44:59` we run TerminateAndReplaceNode (on node `longevity-tls-1tb-7d-longevit-db-node-9bc1e0f0-4`).

then, node-5 is added, and here is node-5 scylla.yaml:
```
api_address: 127.0.0.1
api_doc_dir: /opt/scylladb/api/api-doc/
api_port: 10000
api_ui_dir: /opt/scylladb/swagger-ui/dist/
authenticator: PasswordAuthenticator
authorizer: CassandraAuthorizer
auto_bootstrap: true
batch_size_fail_threshold_in_kb: 50
batch_size_warn_threshold_in_kb: 5
broadcast_rpc_address: 10.0.128.6
cas_contention_timeout_in_ms: 1000
client_encryption_options:
  certificate: /etc/scylla/ssl_conf/client/test.crt
  enabled: true
  keyfile: /etc/scylla/ssl_conf/client/test.key
  truststore: /etc/scylla/ssl_conf/client/catest.pem
cluster_name: longevity-tls-1tb-7d-longevit-db-cluster-9bc1e0f0
commitlog_segment_size_in_mb: 32
commitlog_sync: periodic
commitlog_sync_period_in_ms: 10000
commitlog_total_space_in_mb: -1
endpoint_snitch: org.apache.cassandra.locator.Ec2Snitch
experimental: true
listen_address: 10.0.128.6
murmur3_partitioner_ignore_msb_bits: 12
native_transport_port: 9042
num_tokens: 256
partitioner: org.apache.cassandra.dht.Murmur3Partitioner
read_request_timeout_in_ms: 5000
replace_address_first_boot: 10.0.150.115
rpc_address: 10.0.128.6
rpc_port: 9160
seed_provider:
- class_name: org.apache.cassandra.locator.SimpleSeedProvider
  parameters:
  - seeds: 10.0.153.221
server_encryption_options:
  certificate: /etc/scylla/ssl_conf/db.crt
  internode_encryption: all
  keyfile: /etc/scylla/ssl_conf/db.key
  truststore: /etc/scylla/ssl_conf/cadb.pem
write_request_timeout_in_ms: 2000
```

then, at `2020-04-20 03:48:45` we start the replace_with_repair:
```
< t:2020-04-20 03:47:29,744 f:cluster.py      l:1417 c:sdcm.cluster         p:DEBUG > 2020-04-20T03:47:28+00:00  longevity-tls-1tb-7d-longevit-db-node-9bc1e0f0-5 !INFO    | scylla: [shard 0] repair - Setting max_repair_memory=855008870, max_repair_memory_per_range=33554432, max_repair_ranges_in_parallel=25
< t:2020-04-20 03:48:45,572 f:cluster.py      l:1417 c:sdcm.cluster         p:DEBUG > 2020-04-20T03:48:43+00:00  longevity-tls-1tb-7d-longevit-db-node-9bc1e0f0-5 !INFO    | scylla: [shard 0] repair - replace_with_repair: started with keyspaces={system_traces, system_distributed, mview, keyspace_zstd, keyspace_snappy, system_auth, keyspace1, keyspace_lz4, keyspace_deflate}, source_dc=eu-west
< t:2020-04-20 03:48:45,572 f:cluster.py      l:1417 c:sdcm.cluster         p:DEBUG > 2020-04-20T03:48:43+00:00  longevity-tls-1tb-7d-longevit-db-node-9bc1e0f0-5 !INFO    | scylla: [shard 0] repair - replace_with_repair: started with keyspace=system_traces, source_dc=eu-west, nr_ranges=511
< t:2020-04-20 03:48:45,573 f:cluster.py      l:1417 c:sdcm.cluster         p:DEBUG > 2020-04-20T03:48:43+00:00  longevity-tls-1tb-7d-longevit-db-node-9bc1e0f0-5 !INFO    | scylla: [shard 0] repair - repair id 1 to sync data for keyspace=system_traces, status=started
```

then we see its summary as:
```
[centos@longevity-tls-1tb-7d-longevit-db-node-9bc1e0f0-5 ~]$ grep "repair -" scylla.log | grep replace_with_repair
Apr 20 03:48:43 longevity-tls-1tb-7d-longevit-db-node-9bc1e0f0-5 scylla[2721]:  [shard 0] repair - replace_with_repair: started with keyspaces={system_traces, system_distributed, mview, keyspace_zstd, keyspace_snappy, system_auth, keyspace1, keyspace_lz4, keyspace_deflate}, source_dc=eu-west
Apr 20 03:48:43 longevity-tls-1tb-7d-longevit-db-node-9bc1e0f0-5 scylla[2721]:  [shard 0] repair - replace_with_repair: started with keyspace=system_traces, source_dc=eu-west, nr_ranges=511
Apr 20 03:49:58 longevity-tls-1tb-7d-longevit-db-node-9bc1e0f0-5 scylla[2721]:  [shard 0] repair - replace_with_repair: finished with keyspace=mview, source_dc=eu-west, nr_ranges=764
Apr 20 03:49:58 longevity-tls-1tb-7d-longevit-db-node-9bc1e0f0-5 scylla[2721]:  [shard 0] repair - replace_with_repair: started with keyspace=keyspace_zstd, source_dc=eu-west, nr_ranges=764
Apr 20 04:02:42 longevity-tls-1tb-7d-longevit-db-node-9bc1e0f0-5 scylla[2721]:  [shard 0] repair - replace_with_repair: finished with keyspace=keyspace_zstd, source_dc=eu-west, nr_ranges=764
Apr 20 04:02:42 longevity-tls-1tb-7d-longevit-db-node-9bc1e0f0-5 scylla[2721]:  [shard 0] repair - replace_with_repair: started with keyspace=keyspace_snappy, source_dc=eu-west, nr_ranges=764
Apr 20 04:17:57 longevity-tls-1tb-7d-longevit-db-node-9bc1e0f0-5 scylla[2721]:  [shard 0] repair - replace_with_repair: finished with keyspace=keyspace_snappy, source_dc=eu-west, nr_ranges=764
Apr 20 04:17:57 longevity-tls-1tb-7d-longevit-db-node-9bc1e0f0-5 scylla[2721]:  [shard 0] repair - replace_with_repair: started with keyspace=system_auth, source_dc=eu-west, nr_ranges=764
```

after this, there are messages without mention to `replace_with_repair`:
```
[centos@longevity-tls-1tb-7d-longevit-db-node-9bc1e0f0-5 ~]$ grep "repair -" scylla.log | grep keyspace1 | head -10
Apr 20 03:48:43 longevity-tls-1tb-7d-longevit-db-node-9bc1e0f0-5 scylla[2721]:  [shard 0] repair - replace_with_repair: started with keyspaces={system_traces, system_distributed, mview, keyspace_zstd, keyspace_snappy, system_auth, keyspace1, keyspace_lz4, keyspace_deflate}, source_dc=eu-west
Apr 20 04:18:17 longevity-tls-1tb-7d-longevit-db-node-9bc1e0f0-5 scylla[2721]:  [shard 3] repair - Repair 34 out of 764 ranges, id=7, shard=3, keyspace=keyspace1, table={standard1}, range=(335614119615990384, 340228846832203279]
Apr 20 04:18:17 longevity-tls-1tb-7d-longevit-db-node-9bc1e0f0-5 scylla[2721]:  [shard 8] repair - Repair 34 out of 764 ranges, id=7, shard=8, keyspace=keyspace1, table={standard1}, range=(335614119615990384, 340228846832203279]
Apr 20 04:18:17 longevity-tls-1tb-7d-longevit-db-node-9bc1e0f0-5 scylla[2721]:  [shard 10] repair - Repair 34 out of 764 ranges, id=7, shard=10, keyspace=keyspace1, table={standard1}, range=(335614119615990384, 340228846832203279]
Apr 20 04:18:17 longevity-tls-1tb-7d-longevit-db-node-9bc1e0f0-5 scylla[2721]:  [shard 10] repair - Repair 35 out of 764 ranges, id=7, shard=10, keyspace=keyspace1, table={standard1}, range=(340228846832203279, 346913167480133494]
Apr 20 04:18:17 longevity-tls-1tb-7d-longevit-db-node-9bc1e0f0-5 scylla[2721]:  [shard 3] repair - Repair 35 out of 764 ranges, id=7, shard=3, keyspace=keyspace1, table={standard1}, range=(340228846832203279, 346913167480133494]
Apr 20 04:18:17 longevity-tls-1tb-7d-longevit-db-node-9bc1e0f0-5 scylla[2721]:  [shard 1] repair - Repair 34 out of 764 ranges, id=7, shard=1, keyspace=keyspace1, table={standard1}, range=(335614119615990384, 340228846832203279]
```

those are the repair on the last repair (`repair id 7`):
```
[centos@longevity-tls-1tb-7d-longevit-db-node-9bc1e0f0-5 ~]$ grep "repair -" scylla.log | grep "repair id 7 on shard" | grep "completed"
Apr 20 06:31:44 longevity-tls-1tb-7d-longevit-db-node-9bc1e0f0-5 scylla[2721]:  [shard 10] repair - repair id 7 on shard 10 completed successfully
Apr 20 06:38:42 longevity-tls-1tb-7d-longevit-db-node-9bc1e0f0-5 scylla[2721]:  [shard 12] repair - repair id 7 on shard 12 completed successfully
Apr 20 06:39:46 longevity-tls-1tb-7d-longevit-db-node-9bc1e0f0-5 scylla[2721]:  [shard 11] repair - repair id 7 on shard 11 completed successfully
Apr 20 06:42:06 longevity-tls-1tb-7d-longevit-db-node-9bc1e0f0-5 scylla[2721]:  [shard 0] repair - repair id 7 on shard 0 completed successfully
Apr 20 06:50:26 longevity-tls-1tb-7d-longevit-db-node-9bc1e0f0-5 scylla[2721]:  [shard 9] repair - repair id 7 on shard 9 completed successfully
Apr 20 07:02:02 longevity-tls-1tb-7d-longevit-db-node-9bc1e0f0-5 scylla[2721]:  [shard 8] repair - repair id 7 on shard 8 completed successfully
Apr 20 07:07:17 longevity-tls-1tb-7d-longevit-db-node-9bc1e0f0-5 scylla[2721]:  [shard 13] repair - repair id 7 on shard 13 completed successfully
Apr 20 07:09:24 longevity-tls-1tb-7d-longevit-db-node-9bc1e0f0-5 scylla[2721]:  [shard 3] repair - repair id 7 on shard 3 completed successfully
Apr 20 07:20:40 longevity-tls-1tb-7d-longevit-db-node-9bc1e0f0-5 scylla[2721]:  [shard 5] repair - repair id 7 on shard 5 completed successfully
Apr 20 07:22:59 longevity-tls-1tb-7d-longevit-db-node-9bc1e0f0-5 scylla[2721]:  [shard 2] repair - repair id 7 on shard 2 completed successfully
Apr 20 07:27:07 longevity-tls-1tb-7d-longevit-db-node-9bc1e0f0-5 scylla[2721]:  [shard 4] repair - repair id 7 on shard 4 completed successfully
```

this is the last repair message of node-5:
```
< t:2020-04-20 07:27:16,141 f:cluster.py      l:1417 c:sdcm.cluster         p:DEBUG > 2020-04-20T07:27:07+00:00  longevity-tls-1tb-7d-longevit-db-node-9bc1e0f0-5 !INFO    | scylla: [shard 4] repair - repair id 7 on shard 4 completed successfully
```

cluster is still alive (but node-4, that was terminated):
```
longevity-tls-1tb-7d-longevit-db-node-9bc1e0f0-1 [54.246.227.182 | 10.0.153.221]
longevity-tls-1tb-7d-longevit-db-node-9bc1e0f0-2 [34.245.7.193 | 10.0.166.252]
longevity-tls-1tb-7d-longevit-db-node-9bc1e0f0-3 [34.245.190.21 | 10.0.54.201]
longevity-tls-1tb-7d-longevit-db-node-9bc1e0f0-4 [3.250.2.113 | 10.0.150.115]
longevity-tls-1tb-7d-longevit-db-node-9bc1e0f0-5 [34.244.200.143 | 10.0.128.6]
longevity-tls-1tb-7d-longevit-db-node-9bc1e0f0-6 [34.241.152.147 | 10.0.182.175]
```
monitor node:
```
longevity-tls-1tb-7d-longevit-monitor-node-9bc1e0f0-1 [34.244.173.38 | 10.0.3.235]
```

**Comment by slivne** [comment]
Please provide the cluster info - so we can try and have a look at this

**Comment by slivne** [comment]
@asias please have a look the cluster is still up

**Comment by asias** [comment]
> @asias please have a look the cluster is still up

I need instructions how to access the servers. I tried a few qa ssh keys with no luck and gave up. 

While I can not access the nodes, logs are needed.

**Comment by slivne** [comment]
Access was resolved

**Comment by asias** [comment]
Probably https://github.com/scylladb/scylla/pull/6242 will fix the hang

**Comment by asias** [comment]
I do not need the live cluster now. Can you please collect the logs and delete the cluster.

**Comment by fgelcer** [comment]
> I do not need the live cluster now. Can you please collect the logs and delete the cluster.

i collected the logs, and i'm terminating the cluster:
```
+--------------------------------------------------------------------------------------------------------------------------------------------+
|                                      Collected logs by test-id: 9bc1e0f0-c380-45f7-9733-fe0dc75dc6f8                                       |
+-------------+------------------------------------------------------------------------------------------------------------------------------+
| Cluster set | Link                                                                                                                         |
+-------------+------------------------------------------------------------------------------------------------------------------------------+
| db-cluster  | https://cloudius-jenkins-test.s3.amazonaws.com/9bc1e0f0-c380-45f7-9733-fe0dc75dc6f8/20200421_111431/db-cluster-9bc1e0f0.zip  |
| monitor-set | https://cloudius-jenkins-test.s3.amazonaws.com/9bc1e0f0-c380-45f7-9733-fe0dc75dc6f8/20200421_111431/monitor-set-9bc1e0f0.zip |
| loader-set  | https://cloudius-jenkins-test.s3.amazonaws.com/9bc1e0f0-c380-45f7-9733-fe0dc75dc6f8/20200421_111431/loader-set-9bc1e0f0.zip  |
| sct-runner  | None                                                                                                                         |
+-------------+------------------------------------------------------------------------------------------------------------------------------+
```

**Comment by slivne** [comment]
Fix for #6242 was merged need to see if this reproduces - if not please close

**Comment by slivne** [comment]
repair based operation have been disabled in 4.0.

**Comment by slivne** [comment]
AFAIK 4.2.rc4 is not hitting this or other issues related to row level repair getting stuck - we have fixed many issues up to 4.2.rc4 - if we hit the issue in 4.2.rc4+ please repost and we can reopen
